### ISSUE TITLE
purefb_ds module not updating directory services configuration

### ISSUE THREAD

**Issue by mahendra-maplelabs** [Pure-Storage-Ansible/FlashBlade-Collection#87]
purefb_ds module not updating directory services attributes( URI, base_dn, base_password, bind_user ). Also It will fail when try to remove directory services configuration with flag `state: absent`

1. Create and enable directory service( NFS, SMB, Management ).
2. Try to modify ds attributes( URI, base_dn, base_password, bind_user ).
3. Try to delete directory service configuration using `state: absent`

**Expected behavior**
purefb_ds module should be able to modify existing configuration.

**Desktop (please complete the following information):**
 - purity-fb version v1.9.1
 - collection version v1.3.0
 - Ansible version 2.9.13

**Comment by sdodsley** [comment]
I am not able to reproduce this - please provide the playbook you used and the error output when run with `-vvvv`

**Comment by mahendra-maplelabs** [comment]
Playbook:
```
---
- name: DS configuration
  hosts: localhost
  collections:
    - purestorage.flashblade
  tasks:
  - name: ds | Create Management directory service 
    purefb_ds:
        dstype: management
        enable: true
        state: present
        uri: "ldaps://lab.purestorage.com"
        base_dn: "DC=lab,DC=purestorage,DC=com"
        bind_user: Admin
        bind_password: admin
        fb_url: 10.60.xxx.80
        api_token: T-wergfdsadf-e744-sdfg-bb5a-40d0d565907e
```
Error:
```

TASK [ds | Create Management directory service] **********************************************************************************
task path: /home/mahendra/Projects/Purestorage/pure-ansible/cleanitup/pure-fb-network-setup/test.yml:7
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: mahendra
<127.0.0.1> EXEC /bin/sh -c 'echo ~mahendra && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/mahendra/.ansible/tmp `"&& mkdir "` echo /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905 `" && echo ansible-tmp-1600744009.5-8597-100755325244905="` echo /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905 `" ) && sleep 0'
Using module_utils file ansible_collections/purestorage/flashblade/plugins
Using module_utils file ansible_collections/purestorage
Using module_utils file ansible_collections/purestorage/flashblade
Using module_utils file ansible_collections/purestorage/flashblade/plugins/module_utils/purefb
Using module_utils file ansible_collections
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/basic.py
Using module_utils file ansible_collections/purestorage/flashblade/plugins/module_utils
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/_utils.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/text/formatters.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/validation.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/text/converters.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/pycompat24.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/text/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/six/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/_json_compat.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/_collections_compat.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/parameters.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/parsing/convert_bool.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/parsing/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/compat/selectors.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/_text.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/sys_info.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/process.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/compat/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/file.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/common/collections.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/compat/_selectors2.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/distro/__init__.py
Using module_utils file /home/mahendra/.local/lib/python2.7/site-packages/ansible/module_utils/distro/_distro.py
Using module file /home/mahendra/.ansible/collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py
<127.0.0.1> PUT /home/mahendra/.ansible/tmp/ansible-local-8476iS1Ptz/tmp8eOFtr TO /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905/AnsiballZ_purefb_ds.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905/ /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905/AnsiballZ_purefb_ds.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905/AnsiballZ_purefb_ds.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/mahendra/.ansible/tmp/ansible-tmp-1600744009.5-8597-100755325244905/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_purefb_ds_payload_Ws_7Ky/ansible_purefb_ds_payload.zip/ansible_collections/purestorage/flashblade/plugins/modules/purefb_ds.py", line 161, in enable_ds
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/purity_fb_1dot9/apis/directory_services_api.py", line 511, in update_directory_services
    (data) = self.update_directory_services_with_http_info(directory_service, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/purity_fb_1dot9/apis/directory_services_api.py", line 599, in update_directory_services_with_http_info
    collection_formats=collection_formats)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/api_client/api_client.py", line 331, in call_api
    _return_http_data_only, collection_formats, _preload_content, _request_timeout)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/api_client/api_client.py", line 156, in __call_api
    _request_timeout=_request_timeout)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/api_client/api_client.py", line 392, in request
    body=body)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/api_client/rest.py", line 299, in PATCH
    body=body)
  File "/usr/local/lib/python2.7/dist-packages/purity_fb/api_client/rest.py", line 235, in request
    raise ApiException(http_resp=r)
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "base_dn": "DC=lab,DC=purestorage,DC=com", 
            "bind_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "bind_user": "Admin", 
            "dstype": "management", 
            "enable": true, 
            "fb_url": "10.60.116.80", 
            "join_ou": null, 
            "nis_domain": null, 
            "nis_servers": null, 
            "state": "present", 
            "uri": [
                "ldaps://lab.purestorage.com"
            ]
        }
    }, 
    "msg": "Enable management Directory Service failed"
}

PLAY RECAP ***********************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

**Comment by sdodsley** [comment]
Addressed in #92. 
Please try this PR to see if it resolves your issue.

**Comment by mahendra-maplelabs** [comment]
Fix is working fine. Thanks
